sdk/src/metrics/sync_instruments.cc

```diff
@@ -203,19 +381,122 @@ void DoubleHistogram::Record(double value,
 
 void DoubleHistogram::Record(double value, const opentelemetry::context::Context &context) noexcept
 {
+  if (value < 0)
+  {
+    OTEL_INTERNAL_LOG_WARN(
+        "[DoubleHistogram::Record(V,C)] Value not recorded - negative value for: "
+        << instrument_descriptor_.name_);
+    return;
+  }
   if (!storage_)
   {
+    OTEL_INTERNAL_LOG_WARN(
+        "[DoubleHistogram::Record(V,C)] Value not recorded - invalid storage for: "
+        << instrument_descriptor_.name_);
+    return;
+  }
+  return storage_->RecordDouble(value, context);
+}
+
+#if OPENTELEMETRY_ABI_VERSION_NO >= 2
+void DoubleHistogram::Record(double value,
+                             const opentelemetry::common::KeyValueIterable &attributes) noexcept
+{
+  if (value < 0)
+  {
+    OTEL_INTERNAL_LOG_WARN(
+        "[DoubleHistogram::Record(V,A)] Value not recorded - negative value for: "
+        << instrument_descriptor_.name_);
     return;
   }
-  if (value < 0 || std::isnan(value) || std::isinf(value))
+  if (!storage_)
   {
     OTEL_INTERNAL_LOG_WARN(
-        "[DoubleHistogram::Record(value)] negative/nan/infinite value provided to histogram Name:"
+        "[DoubleHistogram::Record(V,A)] Value not recorded - invalid storage for: "
         << instrument_descriptor_.name_);
     return;
   }
+  auto context = opentelemetry::context::Context{};
+  return storage_->RecordDouble(value, attributes, context);
+}
+
+void DoubleHistogram::Record(double value) noexcept
+{
+  if (value < 0)
+  {
+    OTEL_INTERNAL_LOG_WARN("[DoubleHistogram::Record(V)] Value not recorded - negative value for: "
+                           << instrument_descriptor_.name_);
+    return;
+  }
+  if (!storage_)
+  {
+    OTEL_INTERNAL_LOG_WARN("[DoubleHistogram::Record(V)] Value not recorded - invalid storage for: "
+                           << instrument_descriptor_.name_);
+    return;
+  }
+  auto context = opentelemetry::context::Context{};
   return storage_->RecordDouble(value, context);
 }
+#endif
+
+#if OPENTELEMETRY_ABI_VERSION_NO >= 2
+
+LongGauge::LongGauge(InstrumentDescriptor instrument_descriptor,
+                     std::unique_ptr<SyncWritableMetricStorage> storage)
+    : Synchronous(instrument_descriptor, std::move(storage))
+{
+  if (!storage_)
+  {
+    OTEL_INTERNAL_LOG_ERROR("[LongGauge::LongGauge] - Error during constructing LongGauge."
+                            << "The metric storage is invalid for " << instrument_descriptor.name_);
+  }
+}
+
+void LongGauge::Record(uint64_t value,
+                       const opentelemetry::common::KeyValueIterable *attributes,
+                       const opentelemetry::context::Context *context) noexcept
+{
+  if (!storage_)
+  {
+    OTEL_INTERNAL_LOG_WARN("[LongGauge::Record(V,A,C)] Value not recorded - invalid storage for: "
+                           << instrument_descriptor_.name_);
+    return;
+  }
+  return storage_->RecordLong(value, *attributes, *context);
```

Comment:
Keep in mind attributes and context can be `nullptr` here.

Call the proper `RecordLong()` method, and/or use:

```
auto context = opentelemetry::context::Context{};
```